File: Java/Dagger/atm/src/main/java/ro/razvan/java/dagger/atm/DepositCommandModule.java
package ro.razvan.java.dagger.atm;

import dagger.Binds;
import dagger.Module;
import dagger.multibindings.IntoMap;
import dagger.multibindings.StringKey;

@Module
public interface DepositCommandModule {

    @Binds
    @IntoMap
    @StringKey(DepositCommand.COMMAND_KEY_DEPOSIT)
    Command bindDepositCommand(DepositCommand command);

}

File: Kotlin/Dagger/atm/src/main/kotlin/ro/razvan/kotlin/dagger/atm/DepositCommandModule.kt
package ro.razvan.kotlin.dagger.atm

import dagger.Binds
import dagger.Module
import dagger.multibindings.IntoMap
import dagger.multibindings.StringKey

@Module
interface DepositCommandModule {

    @Binds
    @IntoMap
    @StringKey(COMMAND_KEY_DEPOSIT)
    fun bindDepositCommand(command: DepositCommand): Command

}